pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,732 @@
+	if !releaseExists {
+		return runtime.NewWarningResult("Nextcloud instance not yet ready")
+	}
+
+	svc.Log.Info("Nextcloud release is ready, creating STS, Service, Ingress, Issuer, Certificate, ConfigMap, Job")
+
+	// Add Collabora STS
+	err = AddCollaboraSts(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora STS: " + err.Error())
+	}
+
+	// Add Collabora service
+	err = AddCollaboraService(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora Service: " + err.Error())
+	}
+
+	// Add Collabora ingress
+	err = AddCollaboraIngress(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora Ingress: " + err.Error())
+	}
+
+	// Create install Collabora config map
+	err = createInstallCollaboraConfigMap(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora ConfigMap: " + err.Error())
+	}
+
+	// Create install Collabora job
+	err = createInstallCollaboraJob(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora install Job: " + err.Error())
+	}
+
+	return runtime.NewNormalResult("Collabora deployed and configured!")
+}
+
+func AddCollaboraSts(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+
+	sts := &v1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels: map[string]string{
+				"app": comp.GetName() + "-collabora-code",
+			},
+		},
+		Spec: v1.StatefulSetSpec{
+			Replicas: ptr.To[int32](1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": comp.GetName() + "-collabora-code"},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   comp.GetName() + "-collabora-code",
+					Labels: map[string]string{"app": comp.GetName() + "-collabora-code"},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: comp.GetName() + "-collabora-code-sa",
+					// SecurityContext: &corev1.PodSecurityContext{
+					// 	// https://github.com/CollaboraOnline/online/blob/master/docker/from-packages/Dockerfile#L138
+					// 	RunAsUser: ptr.To[int64](100),
+					// },
+					Containers: []corev1.Container{
+						{
+							Env: []corev1.EnvVar{
+								{
+									Name:  "DONT_GEN_SSL_CERT",
+									Value: "true",
+								},
+								{
+									// https://github.com/nextcloud/all-in-one/discussions/2314
+									// https://github.com/CollaboraOnline/online/issues/3102
+									Name:  "extra_params",
+									Value: "--o:net.proto=IPv4",
+								},
+							},
+							Image: "collabora/code:24.04.8.1.1",
+							Name:  comp.GetName() + "-collabora-code",
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 9980,
+								},
+							},
+							Resources: corev1.ResourceRequirements{
+								Limits: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("4"),
```

Comment:
Please don't hardcode resources, they should be configurable via the component. We don't want to create a new AppCat release just to adjust some limits or requests.

Reply:
Yeah this has to be passed onto component, no doubts. Our component right now almost does just that and nothing else much - adjusting resources :)

Reply:
what should be the approach here?
1) create a SizeSpec in spec.parameters.service.Collabora.plan and create separate Collabora plans?
2) take the same plan as nextcloud is currently running - so if someone need bigger Nextcloud, they receive also bigger collabora
3) create defaults just for collabora?

Reply:
I went with the 3rd option with MariaDB and the ProxySQL.

Reply:
I'll go with 3rd option as well then, I guess it's the easiest one :D

Reply:
It would be better to expose the resources to the customer :). But I guess we can skip for now... We already adjusting pgBouncer resources and ProxySQL will be next I am sure, it's just a matter of time when a customer comes and asks just that